services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/drs/DrsManagerImpl.java

```diff
@@ -58,7 +88,53 @@ public ServiceInformation getServiceInformation() {
         result.setUpdatedAt(UPDATED_AT);
         result.setEnvironment(stackConfiguration.getStack());
         result.setVersion(stackConfiguration.getStackInstance());
-        result.setUrl(DRS_URL);
+        result.setUrl(baseURL);
+        return result;
+    }
+
+    @Override
+    public DrsObject getDrsObject(final Long userId, final String id)
+            throws NotFoundException, DatastoreException, UnauthorizedException, IllegalArgumentException {
+        final UserInfo userInfo = userManager.getUserInfo(userId);
+        final DrsObject result = new DrsObject();
+        final IdAndVersion idAndVersion = IdAndVersion.parse(id);
+        final Entity entity = entityManager.getEntityForVersion(userInfo, idAndVersion.getId().toString(),
+                idAndVersion.getVersion().orElseThrow(() ->
+                        new NotFoundException(String.format("Drs object id %s does not exists", id))), null);
```

Comment:
If the version is missing then throw illegal argument with a message stating that we expect them to include the version (with an example syn123.3).

No need to handle not found special here.